native/src/main/java/io/ballerina/lib/data/csvdata/csv/CsvParser.java

```diff
@@ -334,6 +391,181 @@ public Object execute(Reader reader, Type type, CsvConfig config, BTypedesc bTyp
             }
         }
 
+        private void processFileLogs(Environment environment, AtomicBoolean isOverwritten,
+                                     Exception exception, BMap<?, ?> outputMode, String offendingRow) {
+            boolean enableConsoleLogs = outputMode.getBooleanValue(ENABLE_CONSOLE_LOGS);
+            String dataType = outputMode.getStringValue(DATA_TYPE).toString();
+            if (enableConsoleLogs) {
+                BMap<BString, Object> keyValues = ValueCreator.createMapValue();
+                if (!dataType.equals(METADATA)) {
+                    keyValues.put(OFFENDING_ROW, StringUtils.fromString(offendingRow.trim()));
+                }
+                printErrorLogs(environment, exception, keyValues);
+            }
+            handleFileOutputLogging(environment, outputMode, exception, isOverwritten, offendingRow, dataType);
+        }
+
+        private void processConsoleLogs(Environment environment, Exception exception,
+                                        BMap<?, ?> outputMode, String offendingRow) {
+            boolean excludeSourceData = outputMode.getBooleanValue(EXCLUDE_SOURCE_DATA);
+            BMap<BString, Object> keyValues = ValueCreator.createMapValue();
+            if (excludeSourceData) {
+                keyValues.put(OFFENDING_ROW, StringUtils.fromString(offendingRow.trim()));
+                printErrorLogs(environment, exception, keyValues);
```

Comment:
Can't we call this outside? I believe calling the method with empty key-values works as expected

Reply:
Fixed by [ba54cb5](https://github.com/ballerina-platform/module-ballerina-data.csv/pull/60/commits/ba54cb514a2ee33ed48a4e5c3251c549a82cb120)